arrays/student_validator_using_pointers.c

```diff
@@ -0,0 +1,267 @@
+#include <stdio.h>
+#include <time.h>
+
+typedef struct
+{
+    int day;
+    int month;
+    int year;
+} Date;
+
+typedef struct 
+{
+    Date date_of_birth;
+    char sex;
+} PersonDate;
+
+int validate_input(PersonDate* card)
+{
+    return 
+    !(
+        &(*card).date_of_birth.year < 1901 || 
+        &(*card).date_of_birth.day > 31 || 
+        &(*card).date_of_birth.month > 12 || 
+        &(*card).date_of_birth.day < 1 || 
+        &(*card).date_of_birth.month < 1
+    ) || (
+        &(*card).sex != 'M' && 
+        &(*card).sex != 'F'
+    );
```

Comment:
Try to make this condition more readable. Try decompose the big condition into several smaller ones, store the smaller ones in separate variables or functions. Try to use `&&` instead of `||` where possible